tests/components/watergate/test_init.py

```diff
@@ -0,0 +1,71 @@
+"""Tests for the Watergate integration init module."""
+
+from unittest.mock import patch
+
+from homeassistant.components.watergate.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from .const import MOCK_WEBHOOK_ID
+
+from tests.common import ANY
+from tests.typing import ClientSessionGenerator
+
+
+async def test_async_setup_entry(
+    hass: HomeAssistant, mock_entry, mock_watergate_client
+) -> None:
+    """Test setting up the Watergate integration."""
+    hass.config.components.add("network")
+    hass.config.internal_url = "http://hassio.local"
+    mock_entry.add_to_hass(hass)
+
+    with (
+        patch("homeassistant.components.webhook.async_register") as mock_webhook,
+    ):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()  # Wait for all setup tasks to complete
+
+        assert mock_entry.state is ConfigEntryState.LOADED
+
+        mock_webhook.assert_called_once_with(
+            hass,
+            DOMAIN,
+            "Watergate",
+            MOCK_WEBHOOK_ID,
+            ANY,
+        )
+        mock_watergate_client.async_set_webhook_url.assert_called_once_with(
+            f"http://hassio.local/api/webhook/{MOCK_WEBHOOK_ID}"
+        )
+        mock_watergate_client.async_get_device_state.assert_called_once()
+
+
+async def test_handle_webhook(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    mock_entry,
+    mock_watergate_client,
+) -> None:
+    """Test handling webhook events."""
+    hass.config.components.add("network")
+
+    mock_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(mock_entry.entry_id)
+    await hass.async_block_till_done()  # Wait for all setup tasks to complete
+
+    entity_id = "valve.sonic"
+
+    assert hass.states.get(entity_id).state == "open"
```

Comment:
assert state first, and use enums

Reply:
What do you mean by 'assert state first'?

Reply:
I know it already.